lib/rubocop/cop/rails/migration_file_name.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop makes sure that each migration file defines a migration class
+      # whose name matches the file name.
+      # (i.e. `20220224111111_create_users.rb` should define `CreateUsers` class.)
+      #
+      # @example
+      #   # db/migrate/20220224111111_create_users.rb
+      #
+      #   # bad
+      #
+      #   class SellBooks < ActiveRecord::Migration[7.0]
+      #   end
+      #
+      #   # good
+      #
+      #   class CreateUsers < ActiveRecord::Migration[7.0]
+      #   end
+      #
+      class MigrationFileName < Base
+        MSG = 'The class name does not match with the file name.'
+
+        def on_new_investigation
+          filepath = processed_source.file_path
+          @basename = File.basename(filepath, '.rb')
+        end
```

Comment:
Can you move these logic in the `on_class` method and refactor to prevent the use of the instance variable `@basename`?
https://refactoring.com/catalog/inlineFunction.html